src/components/app/details/appDetails/DeploymentStatusDetailRow.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { useEffect, useState } from 'react'
+import moment from 'moment'
+import { useParams } from 'react-router-dom'
+import { DEPLOYMENT_STATUS, MANIFEST_STATUS_HEADERS, Moment12HourFormat, TERMINAL_STATUS_MAP, TIMELINE_STATUS } from '../../../../config'
+import { showError } from '../../../common'
+import { ShowMoreText } from '../../../common/ShowMoreText'
+import { getManualSync } from '../../service'
+import { DeploymentStatusDetailRowType } from './appDetails.type'
+import { ReactComponent as Close } from '../../../../assets/icons/ic-close.svg'
+import { ReactComponent as Check } from '../../../../assets/icons/ic-check.svg'
+import { ReactComponent as Timer } from '../../../../assets/icons/ic-timer.svg'
+import { ReactComponent as Question } from '../../../../assets/icons/ic-help-outline.svg'
+import { ReactComponent as Error } from '../../../../assets/icons/ic-error-exclamation.svg'
+import { ReactComponent as Disconnect } from '../../../../assets/icons/ic-disconnected.svg'
+import { ReactComponent as DropDownIcon } from '../../../../assets/icons/appstatus/ic-chevron-down.svg'
+import { ReactComponent as TimeOut } from '../../../../assets/icons/ic-timeout-red.svg'
+import AppStatusDetailsChart from '../../../v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailsChart'
+
+export function DeploymentStatusDetailRow({
+    type,
+    hideVerticalConnector,
+    deploymentDetailedData,
+    streamData,
+}: DeploymentStatusDetailRowType) {
+    const { appId, envId } = useParams<{ appId: string; envId: string }>()
+    const [collapsed, toggleCollapsed] = useState<boolean>(
+        deploymentDetailedData.deploymentStatusBreakdown[type].isCollapsed,
```

Comment:
Can we keep the data for `deploymentDetailedData.deploymentStatusBreakdown[type]` in some var at the top & then use it everywhere?

Reply:
done